apps/cowswap-frontend/src/common/utils/assertProviderNetwork.ts

```diff
@@ -0,0 +1,23 @@
+import { getChainIdImmediately } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { JsonRpcProvider, Provider } from '@ethersproject/providers'
+
+export async function assertProviderNetwork(
+  chainId: SupportedChainId,
+  provider: JsonRpcProvider | Provider,
+  description: string,
+): Promise<SupportedChainId> {
+  const ethereumProvider = (provider as unknown as { provider: typeof window.ethereum }).provider
```

Comment:
I know, it looks a bit ugly, but I don't want to pull to many dependencies for this check